### PR TITLE
GLT-850: attempt to generate non-standard migration aliases

### PR DESCRIPTION
**Pull request by djcooke**

### CODE REVIEW

File: migration/src/main/java/uk/ac/bbsrc/tgac/miso/migration/util/OicrMigrationNamingScheme.java [hunk 1]
```diff
@@ -44,4 +56,39 @@ private ValidationResult validateAdditionalPatterns(String value) {
 
   }
 
+  private static class OicrMigrationSampleAliasGenerator extends OicrSampleAliasGenerator {
+
+    @Override
+    public String generate(Sample sample) throws MisoNamingException {
+      // attempt to ignore up to two levels of non-standard aliases
```
Comment (apmasell) [inline migration/src/main/java/uk/ac/bbsrc/tgac/miso/migration/util/OicrMigrationNamingScheme.java]: Why two?

Comment (djcooke) [inline migration/src/main/java/uk/ac/bbsrc/tgac/miso/migration/util/OicrMigrationNamingScheme.java]: The second level is needed for MDT migration. Depending on child class and hierarchy, the child portion of the alias is appended to either the parent's or the grandparent's alias.

For example, a Stock's alias uses the Tissue alias as a base, but there may be a TissueProcessing in the middle. Aliquot aliases also use the Tissue alias as a base, and that should always be their grandparent. A third level could potentially be needed (multiple TissueProcessing levels), but it's unlikely, so not bothering for now

Comment (apmasell) [inline migration/src/main/java/uk/ac/bbsrc/tgac/miso/migration/util/OicrMigrationNamingScheme.java]: Okay. Please paste that in a comment.

Comment (djcooke) [inline migration/src/main/java/uk/ac/bbsrc/tgac/miso/migration/util/OicrMigrationNamingScheme.java]: Updated comment

**Comment by djcooke** [comment]
Added check to throw exception if sample somehow still saved with null alias

**Pull request merged**